### DISCUSSION TITLE
Loading ABIs from JSON files

### DISCUSSION THREAD

**Discussion by chmanie** [wevm/viem#1009] [Question] [answered]
The ABIs I would like to use come as [Truffle](https://trufflesuite.com/) JSON artifacts. In the [docs](https://viem.sh/docs/typescript.html#type-inference) I saw the following tip:

> Unfortunately [TypeScript doesn't support importing JSON as const](https://github.com/microsoft/TypeScript/issues/32063). Check out [@wagmi/cli](https://wagmi.sh/cli) to help with this! It can automatically fetch ABIs from Etherscan, resolve ABIs from your Foundry/Hardhat projects, and much more.

I can't find a way to generate ABIs from JSON files using `@wagmi/cli`. What would be the canonical way to achieve this?

The only way this would work is by casting all the JSON abis as `any`:
```ts
import { defineConfig } from '@wagmi/cli';

import { abi } from './ABI.json';

export default defineConfig({
  out: 'src/generated.ts',
  contracts: [{ name: 'MyContract', abi: abi as any }],
  plugins: [],
});
```

**Comment by tmm** [comment, marked as answer]
Casting to `any` or `Abi` is perfectly reasonable to me!
